src/praveen/vehicleDirection.ts

```diff
@@ -0,0 +1,86 @@
+enum Move {
+  STRAIGHT = 'STRAIGHT',
+  LEFT = 'LEFT',
+  RIGHT = 'RIGHT',
+  BACK = 'BACK'
+}
+
+enum Direction {
+  NORTH = 'NORTH',
+  EAST = 'EAST',
+  WEST = 'WEST',
+  SOUTH = 'SOUTH'
+}
+
+enum PossibleMovesFromNorth {
+  STRAIGHT = Direction.NORTH,
+  RIGHT = Direction.EAST,
+  LEFT = Direction.WEST,
+  BACK = Direction.SOUTH
+}
+
+enum PossibleMovesFromEast {
+  STRAIGHT = Direction.EAST,
+  RIGHT = Direction.SOUTH,
+  LEFT = Direction.NORTH,
+  BACK = Direction.WEST
+}
+
+enum PossibleMovesFromSouth {
+  STRAIGHT = Direction.SOUTH,
+  RIGHT = Direction.WEST,
+  LEFT = Direction.EAST,
+  BACK = Direction.NORTH
+}
+
+enum PossibleMovesFromWest {
+  STRAIGHT = Direction.WEST,
+  RIGHT = Direction.NORTH,
+  LEFT = Direction.SOUTH,
+  BACK = Direction.EAST
+}
+
+/**
+ * Function to return next direction based on the move
+ * @param {string} currentDirection
+ * @param {string} nextMove
+ * @return {Direction} next direction based on the move
+ */
+const getNextDirectionForMove = (
+  currentDirection: string,
+  nextMove: string
+): Direction => {
+  let nextDirection: Direction;
+  if (currentDirection === 'NORTH') {
```

Comment:
Convert multiple if else block to switch case

Reply:
fixed